zapcore/encoder.go

```diff
@@ -20,6 +20,21 @@
 
 package zapcore
 
+import "time"
+
+// An EncoderConfig allows users to configure the concrete encoders supplied by
+// zapcore.
+type EncoderConfig struct {
+	// Set the keys used for each log entry.
+	MessageKey, LevelKey, TimeKey, NameKey, CallerKey, StacktraceKey string
+	// Configure the primitive representations of common complex types. For
+	// example, some users may want all time.Times serialized as floating-point
+	// seconds since epoch, while others may prefer ISO8601 strings.
+	LevelFormatter    func(Level, ArrayEncoder)
```

Comment:
`EncodeFoo` sounds good. `LevelEncoder` sounds like `JSONEncoder`, but is a whole different beast.